Mixpanel/Mixpanel.m

```diff
@@ -563,14 +562,15 @@ - (void)reset
         self.superProperties = [NSMutableDictionary dictionary];
         self.people.distinctId = nil;
         self.people.unidentifiedQueue = [NSMutableArray array];
-        self.eventsQueue = [NSMutableArray array];
-        self.peopleQueue = [NSMutableArray array];
         self.timedEvents = [NSMutableDictionary dictionary];
         self.shownSurveyCollections = [NSMutableSet set];
         self.shownNotifications = [NSMutableSet set];
         self.decideResponseCached = NO;
         self.variants = [NSSet set];
         self.eventBindings = [NSSet set];
+        
+        [self archiveAllEvents:[NSMutableArray array]];
+        [self archiveAllPeople:[NSMutableArray array]];
```

Comment:
Seems that every time we call `[self archive]` we always call `[self archiveAllEvents:]` and `[self archiveAllPeople:]`. Looks reasonable to me to include `archiveAllEvents` and `archiveAllPeople` inside `archive` and we pass the `NSMutableArray` as params. What do you think? If we do that, we can modify this line and `flushWithCompletionHandler` method as well.